improver_tests/threshold/test_ThresholdPytest.py

```diff
@@ -0,0 +1,599 @@
+                        [
+                            np.r_[[1] * 2, [0] * 3, [1] * 2, [0] * 18]
+                            .reshape((5, 5))
+                            .astype(np.float32),
+                            np.r_[[1] * 3, [0] * 2, [1] * 3, [0] * 2, [1] * 3, [0] * 12]
+                            .reshape((5, 5))
+                            .astype(np.float32),
+                        ]
+                    ),
+                    np.stack(
+                        [
+                            np.zeros((5, 5), dtype=np.float32),
+                            np.zeros((5, 5), dtype=np.float32),
+                        ]
+                    ),
+                ]
+            ),
+        ),
+        # Multiple thresholds applied.
+        (
+            {"threshold_values": [0.5, 1.5]},
+            1,
+            np.ones((3, 3)),
+            np.stack(
+                [np.ones((3, 3), dtype=np.float32), np.zeros((3, 3), dtype=np.float32)]
+            ),
+        ),
+        # threshold value units are specified and differ from the data; if units matches
+        # the returned probabilities would be 1, but taking into account the units
+        # the returned probabilities are 0.
+        (
+            {"threshold_values": 0.5, "threshold_units": "m/s"},
+            1,
+            np.ones((3, 3)),
+            np.zeros((3, 3), dtype=np.float32),
+        ),
+    ],
+)
+def test_bespoke_expected_values(custom_cube, kwargs, expected_result):
+    """Test that thresholding yields the expected data values for
+    different configurations. Variations tried here are:
+
+      - Different threshold values relative to the diagnostic value(s)
+      - Use of fuzzy thresholds, specified in different ways.
+      - Deterministic, single realization, and multi-realization inputs
+      - Different threshold comparators. Note that the tests have been
+        engineered such that there are no cases where the difference
+        between "ge" and "gt", or "le" and "lt" are signficant, these
+        are tested elsewhere.
+      - Collapsing and not collapsing the realization coordinate when
+        present.
+    """
+    plugin = Threshold(**kwargs)
+    result = plugin(custom_cube)
+
+    assert result.data.shape == expected_result.shape
+    assert np.allclose(result.data, expected_result)
+    assert type(result.data) == type(expected_result)
+    assert result.data.dtype == expected_result.dtype
+
+
+@pytest.mark.parametrize(
+    "kwargs,n_realizations,data,mask,expected_result",
+    [
+        # the land corner has a value of 1, this is not spread across the
+        # other points within the vicinity as these are sea points.
+        (
+            {"threshold_values": 0.5, "vicinity": 3000},
+            1,
+            np.array([[0, 0, 1.0], [0, 0, 0], [0, 0, 0]]),
+            np.array([[0, 0, 1], [0, 0, 0], [0, 0, 0]]),
+            np.array([[0, 0, 1.0], [0, 0, 0], [0, 0, 0]], dtype=np.float32),
```

Comment:
Please can we have consistent values of 1 throughout? Either all `1` or all `1.0`, not a mixture.